sympy/core/power.py

```diff
@@ -579,9 +580,12 @@ def _check(ct1, ct2, old):
                     pow = as_int(pow)
                     combines = True
                 except ValueError:
-                    combines = Pow._eval_power(
-                        Pow(*old.as_base_exp(), evaluate=False),
-                        pow) is not None
+                    if denom(pow) == 1:
```

Comment:
use of `denom(2*pi)` is `1` is to check if the final exponent is going to be fraction or not. I am unable to understand your last line ` I am not sure but in try block it should check is_imaginary, is_number and as_int for pow around line 580.`